backend/store/postgres/sync.go

```diff
@@ -0,0 +1,110 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/sensu/sensu-go/backend/store"
+)
+
+// SynchronizedExecutor functions as a postgresql advisory lock based mutex.
+type SynchronizedExecutor struct {
+	DB DBI
+
+	CheckinInterval time.Duration
+}
+
+// Execute blocks until the Mutex can be locked, executes the handler function,
+// then unlocks the Mutex. Implemented with postgresql advisory locks.
+func (se *SynchronizedExecutor) Execute(ctx context.Context, mux store.Mutex, handler store.MutexHandler) error {
+	var next time.Time
+	for {
+		now := time.Now()
+		if next.After(now) {
+			select {
+			case <-ctx.Done():
+				return nil
+			case <-time.After(next.Sub(now)):
+			}
+		}
+		tx, err := se.tryLock(ctx, mux)
+		if err != nil {
+			logger.WithError(err).Errorf("error trying to lock mutex: %d", mux)
+			continue
+		}
+		if tx != nil {
+			defer func() {
+				if err := tx.Rollback(ctx); err != nil {
```

Comment:
I think this is valid, but I have experienced customer support issues with unconditional rollbacks, as I have also used them in the past. The reason is that rollbacks are things that postgresql operators look for, as signs of problems. If the system is malfunctioning for other reasons, it can be a false signal. Is it necessary to rollback here, or would commit have the same effect?

Reply:
It's my understanding that the difference between rollback would be negligible.  Will change these to Commits.